server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -334,6 +336,18 @@ public ClusterState execute(ClusterState currentState) {
                         );
                     }
                 }
+
+                boolean remoteStoreIndexShallowCopy = REMOTE_STORE_INDEX_SHALLOW_COPY.get(repository.getMetadata().settings());
+                Version minVersionInCluster = currentState.nodes().getMinNodeVersion();
+                if (remoteStoreIndexShallowCopy && !minVersionInCluster.onOrAfter(Version.V_2_9_0)) {
+                    remoteStoreIndexShallowCopy = false;
+                    logger.warn(
+                        "Setting {} is ignored as some of the nodes in cluster are on version older than {}. Minimum node version in cluster is: {}",
+                        REMOTE_STORE_INDEX_SHALLOW_COPY,
+                        Version.V_2_9_0,
+                        minVersionInCluster
+                    );
```

Comment:
Lets fail it rather than silently absorbing the incompatible settings during mixed mode

Reply:
ack. now added validation during put repository, updating repository settings itself will fail in case of mixed cluster mode.